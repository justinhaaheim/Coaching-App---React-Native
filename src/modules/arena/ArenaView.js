import React, { Component, PropTypes } from 'react';
import { View } from 'react-native';
import { Button, Container, Content, H3, Text } from 'native-base';

import { arena } from '../../config/data';

class ArenaView extends Component {
  static navigationOptions = {
    title: 'Coaching Arena',
  };

  static propTypes = {
    mleEnabled: PropTypes.bool.isRequired,
    arenaStateActions: PropTypes.shape({
      toggleButton: PropTypes.func.isRequired,
      resetArena: PropTypes.func.isRequired,
    }).isRequired,
    qualitiesList: PropTypes.array.isRequired,
    // TODO: specify exact shape of buttons array
    buttons: PropTypes.array.isRequired,
  };

  formatQualitiesList(list) {
    const listLower = list.map(s => s.toLowerCase());
    if (listLower.length === 0) {
      return '___';
    } else if (listLower.length === 1) {
      return listLower[0];
    }
    const last = listLower.pop();
    return `${listLower.join(', ')} and ${last}`;
  }

  render() {
    const questions = this.props.mleEnabled ? arena.questions.mle : arena.questions.client;
    const { toggleButton, resetArena } = this.props.arenaStateActions;
    const { buttons } = this.props;

    const buttonComponents = buttons.map((b, index) =>
      (<View style={styles.buttonWrapper} key={b.name}>
        <Button
          key={index}
          light={!b.selected}
          dark={b.selected}
          small
          onPress={() => toggleButton(index)}
        >
          <Text>
            {b.name}
          </Text>
        </Button>
      </View>),
    );

    return (
      <Container style={styles.windowContainer}>
        <Content style={styles.contentContainer}>
          <H3 style={styles.h3}>Qualities of Being</H3>
          <View
            style={{
              flex: 1,
              flexDirection: 'row',
              justifyContent: 'flex-start',
              marginBottom: 20,
            }}
          >
            <View
              style={{
                flexDirection: 'column',
              }}
            >
              {buttonComponents.slice(0, 12)}
            </View>
            <View
              style={{
                flexDirection: 'column',
              }}
            >
              {buttonComponents.slice(12)}
              <View style={styles.buttonWrapper}>
                <Button warning small onPress={() => resetArena()}>
                  <Text>Reset</Text>
                </Button>
              </View>
            </View>
          </View>

          <H3 style={styles.h3}>Arena Questions</H3>
          <Text style={styles.bodyText}>
            1) {questions[1]}
          </Text>
          <Text style={styles.bodyTextItalic}>
            {`I am willing to be ${this.formatQualitiesList(this.props.qualitiesList)}.`}
          </Text>
          <Text style={styles.bodyText}>
            2) {questions[2]}
          </Text>
          <Text style={styles.bodyText}>
            3) {questions[3]}
          </Text>
          <Text style={styles.bodyText}>
            4) {questions[4]}
          </Text>
          <Text style={styles.copyright}>
            {`The Coaching Arena Â© 1997-${new Date().getFullYear()} The Academy for Coaching Excellence`}
          </Text>
        </Content>
      </Container>
    );
  }
}

const styles = {
  buttonWrapper: {
    margin: 3,
  },
  windowContent: {
    backgroundColor: 'white',

  },
  windowContainer: {
    backgroundColor: 'white',
  },
  contentContainer: {
    padding: 20,
  },
  h3: {
    marginTop: 12,
    marginBottom: 8,
  },
  bodyText: {
    fontSize: 16,
    marginBottom: 8,
  },
  bodyTextItalic: {
    fontSize: 16,
    marginBottom: 8,
    fontStyle: 'italic',
  },
  copyright: {
    fontSize: 16,
    fontStyle: 'italic',
    marginVertical: 12,
  },
};

export default ArenaView;
