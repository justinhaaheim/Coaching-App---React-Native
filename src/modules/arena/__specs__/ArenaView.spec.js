import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import ArenaView from '../ArenaView';

test('renders correctly', () => {
  const tree = renderer
    .create(
      <ArenaView
        mleEnabled
        arenaStateActions={{
          toggleButton: () => 'Toggle Button',
          resetArena: () => 'Reset Arena',
        }}
        qualitiesList={['Test Word']}
        buttons={[{ name: 'Test Button', selected: false }]}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
