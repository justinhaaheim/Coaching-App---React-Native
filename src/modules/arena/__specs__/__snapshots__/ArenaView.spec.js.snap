// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "height": 1334,
    }
  }
  virtual={undefined}
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutoAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardDismissMode="interactive"
    onScroll={[Function]}
    padder={true}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={0}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      }
    }
    virtual={undefined}
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontSize": 21,
            "lineHeight": 22,
            "marginBottom": 8,
            "marginTop": 18,
          }
        }
        virtual={undefined}
      >
        QUALITIES OF BEING
      </Text>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginBottom": 20,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <View
            style={
              Object {
                "margin": 3,
              }
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              isTVSelectable={true}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "flex-start",
                  "backgroundColor": "#f4f4f4",
                  "borderBottomWidth": null,
                  "borderColor": "#007aff",
                  "borderLeftWidth": null,
                  "borderRadius": 5,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 30,
                  "justifyContent": "space-between",
                  "opacity": 1,
                  "paddingBottom": 6,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 6,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "lineHeight": 19,
                    "marginLeft": 0,
                    "marginRight": 0,
                  }
                }
                virtual={undefined}
              >
                Test Button
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <View
            style={
              Object {
                "margin": 3,
              }
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              isTVSelectable={true}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "flex-start",
                  "backgroundColor": "#f0ad4e",
                  "borderBottomWidth": null,
                  "borderColor": "#007aff",
                  "borderLeftWidth": null,
                  "borderRadius": 5,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 30,
                  "justifyContent": "space-between",
                  "opacity": 1,
                  "paddingBottom": 6,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 6,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#fff",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "lineHeight": 19,
                    "marginLeft": 0,
                    "marginRight": 0,
                  }
                }
                virtual={undefined}
              >
                Reset
              </Text>
            </View>
          </View>
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontSize": 21,
            "lineHeight": 22,
            "marginBottom": 8,
            "marginTop": 18,
          }
        }
        virtual={undefined}
      >
        ARENA QUESTIONS
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 15,
            "marginBottom": 8,
          }
        }
        virtual={undefined}
      >
        1) 
        Who am I willing to be in order to produce an extraordinary result out of this interaction?
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 15,
            "fontStyle": "italic",
            "marginBottom": 8,
          }
        }
        virtual={undefined}
      >
        I am willing to be test word.
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 15,
            "marginBottom": 8,
          }
        }
        virtual={undefined}
      >
        2) 
        Am I willing to systematically dismantle my structure of knowing?
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 15,
            "marginBottom": 8,
          }
        }
        virtual={undefined}
      >
        3) 
        Am I willing to be a demand for coaching?
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 15,
            "marginBottom": 8,
          }
        }
        virtual={undefined}
      >
        4) 
        Am I willing to guarantee that whoever coaches me will be successful?
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 15,
            "fontStyle": "italic",
            "marginVertical": 12,
          }
        }
        virtual={undefined}
      >
        The Coaching Arena Â© 1997-2017 The Academy for Coaching Excellence
      </Text>
    </View>
  </RCTScrollView>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    supportedOrientations={null}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-start",
          "margin": 20,
        }
      }
    >
      <View
        danger={false}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0,0,0,0.8)",
            "borderRadius": 5,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "minHeight": 50,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          }
        }
        success={false}
        virtual={undefined}
        warning={false}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "flex": 1,
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          virtual={undefined}
        />
      </View>
    </View>
  </Modal>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,0.4)",
          "flex": 1,
          "justifyContent": "flex-end",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "elevation": 4,
            "height": NaN,
            "opacity": 1,
            "padding": 15,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#757575",
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          virtual={undefined}
        />
        <RCTScrollView
          dataArray={Array []}
          dataSource={
            ListViewDataSource {
              "items": 0,
            }
          }
          enableEmptySections={true}
          renderRow={[Function]}
          renderScrollComponent={[Function]}
          style={
            Object {
              "marginHorizontal": -15,
              "marginTop": 15,
            }
          }
          virtual={undefined}
        >
          <View />
        </RCTScrollView>
      </View>
    </View>
  </Modal>
</View>
`;
